vmm/src/vm.rs

```diff
@@ -980,12 +984,55 @@ impl Vm {
         let new_state = VmState::Running;
         current_state.valid_transition(new_state)?;
 
+        // On x86_64, load_kernel() invokes configure_system().
+        // But on aarch64, it only loads kernel, configure_system()
+        // need to be postponed after VCPU's are created.
         let entry_addr = self.load_kernel()?;
 
+        // create and configure vcpus
+        self.cpu_manager
+            .lock()
+            .unwrap()
+            .create_boot_vcpus(entry_addr)
+            .map_err(Error::CpuManager)?;
+
+        //
+        // The commented code below illustrates why we have to separate
+        // VCPU's creation and activation into 2 stages to cover aarch64.
+        // (It will be de-commented when supporting code is ready.)
+        //
+        // On aarch64, VCPU's have to be created in prior of GIC.
+        //
+        // #[cfg(target_arch = "aarch64")]
+        // let boot_vcpus = self.cpu_manager.lock().unwrap().boot_vcpus();
+        //
+        // #[cfg(target_arch = "aarch64")]
+        // self.device_manager
+        //     .lock()
+        //     .unwrap()
+        //     .create_irqchip(boot_vcpus)
```

Comment:
Are you planning on using the GICv2 created by KVM through this call? Or is there some equivalent to SPLIT_IRQCHIP for ARM?

Reply:
I used a misleading name `create_irqchip()`. Here I was not referring to the the KVM API `create_irqchip` for creating GICv2. 

By the way, the creation of  GIC will be placed in `configure_system()`, the commented code will never be present. We will always try to create a GICv3 at first, only retry for GICv2 when hardware doesn't support.